nightshift/marc_parser.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module reads and parses MARC Sierra records (bibliographic and order data)
+to be inserted into the DB.
+In case of e-resouces only bib informatin is considered.
+Source MARC files for e-resources will have a mix of various formats (ebooks, eaudio,
+evideo)
+"""
+# from collections import namedtuple
+import pickle
+from typing import Any, Iterator, Optional
+
+from bookops_marc import SierraBibReader, Bib
+
+
+from .constants import LIBRARIES, RESOURCE_CATEGORIES
+from .datastore import Resource
+
+
+class BibReader:
+    def __init__(
+        self,
+        marc_fh: str,
+        library: str,
+        hide_utf8_warnings: bool = True,
+    ) -> None:
+        """
+        Iterator that yields bib information extracted from MARC records
+        Args:
+            marc_fh:                        MARC file's path
+            library:                        'nyp' or 'bpl'
+            hide_utf8_warnings:             hides character encoding warnings
+        """
+
+        if library not in LIBRARIES.keys():
+            raise ValueError("Invalid 'library' argument. Must be 'nyp' or 'bpl'.")
+
+        self.marc_fh = marc_fh
+        self.library = library
+        self.hide_utf8_warnings = hide_utf8_warnings
+
+    def __iter__(self) -> Iterator[Resource]:
+        with open(self.marc_fh, "rb") as marcfile:
+            reader = SierraBibReader(
+                marcfile, hide_utf8_warnings=self.hide_utf8_warnings
+            )
+            for bib in reader:
+                resource_category = self._determine_resource_category(bib)
+
+                # skip any unmapped resource types from processing
+                if not resource_category:
+                    continue
+                else:
+                    bib_info = self._map_data(bib, resource_category)
+                    yield bib_info
+                    # log a warning
+
+    def _determine_resource_category(self, bib: Bib) -> Optional[str]:
+        """
+        Determies resource category based on bib and order information.
+
+        Args:
+            bib:                            `bookops_marc.Bib` instance
+
+        Returns:
+            resource_category
+        """
+        # Overdrive MarcExpress records control number starts with ODN
```

Comment:
I'm assuming that everything we're dealing with now is from overdrive, but in the interest of making this extensible in the future, maybe this should be made configurable via an environment variable or other means (constants file)? If it took an array and checked all of them we could be sure that we are future-proofing this